### PR TITLE
inventory: fix black screen frames between inventory mode

### LINKED ISSUE #75: OG bugfix: blinking when switching between inventory modes
When navigating from the main menu ring to the keys or the options or back, the game renders a black screen briefly.

### PR DESCRIPTION
**Pull request by walkawayy**
Resolves #75.

I don't know much about rendering, so let me know if this is an acceptable fix or not. If it's not, I hope this can help you guys figure out a different solution.

### CODE REVIEW

**Comment by rr-** [comment]
I think this is acceptable but now I don't think we need the calls to `GFX_Context_SetRendered` in `GFX_3D_Renderer_RenderPrimList` and `GFX_3D_Renderer_RenderPrimStrip`.

**Review by rr- — APPROVED** [review]
See above, + we need to remove the comment

**Comment by walkawayy** [comment]
Ok definitely let me know if this is what you suggested. The game still seems to perform fine?

**Comment by walkawayy** [comment]
Actually the one thing I noticed is the Eidos screen at the game's boot up seems to show for only a split second instead of the normal length.

File: src/gfx/2d/2d_surface.c [hunk 1]
```diff
@@ -165,13 +165,9 @@ bool GFX_2D_Surface_Flip(GFX_2D_Surface *surface)
...
     }
 
-    bool rendered = GFX_Context_IsRendered();
-
     // don't re-upload surfaces if external rendering was active after
     // lock() has been called, since it wouldn't be visible anyway
-    if (rendered) {
-        surface->is_dirty = false;
-    }
+    surface->is_dirty = false;
```
Comment (rr-) [inline src/gfx/2d/2d_surface.c:170]: The comment above no longer makes sense and can be removed.

File: src/gfx/2d/2d_surface.c [hunk 2]
```diff
@@ -191,23 +187,14 @@ bool GFX_2D_Surface_Flip(GFX_2D_Surface *surface)
 
     // swap buffer now if there was external rendering, otherwise the
     // surface would overwrite it
-    if (rendered) {
-        GFX_Context_SwapBuffers();
-    }
+    GFX_Context_SwapBuffers();
```
Comment (rr-) [inline src/gfx/2d/2d_surface.c:190]: The comment above no longer makes sense and can be removed.

**Comment by rr-** [comment]
Actually let's revert my earlier suggestions, I understand GLRage's approach. I can't think of anything better at the moment without complicating the code more than it is. 

One thing I'd like is to avoid making `game` depend directly on `gfx`. To that end I suggest adding a new routine, `void GFX_3D_Renderer_RenderEmpty()` that calls `GFX_Context_SetRendered`. Proxy it through `S_Output_DrawEmpty()` and finally `Output_DrawEmpty` to trigger an "external redraw" in GLRage's terminology.

**Comment by walkawayy** [comment]
Ok cool. I'm going to push up a clean branch and PR with that different approach.

**Pull request closed without merging**